README.md

```diff
@@ -137,3 +138,99 @@ fn main() -> Result<(), RuntimeError> {
 }
 ```
 
+### Build Instructions
+
+#### Building `wamr-sys` and `wamr-rust-sdk`
+
+To build the `wamr-sys` and `wamr-rust-sdk` crates, follow these steps:
+
+1. Ensure you have the Rust toolchain installed.
+2. Clone the repository:
+
+   ```sh
+   git clone https://github.com/bytecodealliance/wamr-rust-sdk.git
+   cd wamr-rust-sdk
+   ```
+
+3. Build the `wamr-sys` crate:
+
+   ```sh
+   cargo build -p wamr-sys
+   ```
+
+4. Build the `wamr-rust-sdk` crate:
+
+   ```sh
+   cargo build
+   ```
+
+#### Preparing a Development and Building Environment
+
+##### For non-espidf targets
+
+1. Prepare the Rust toolchain:
+
+   ```sh
+   rustup default stable
+   rustup target add wasm32-unknown-unknown
+   rustup target add wasm32-wasip1
+   rustup component add clippy
+   rustup component add rustfmt
+   ```
+
+2. If targeting a non-linux platform, set `WAMR_BUILD_TARGET` and `WAMR_BUILD_PLATFORM` in the `.cargo/config.toml`:
+
+   ```toml
+   [env]
+   WAMR_BUILD_PLATFORM = "OS name"
+   WAMR_BUILD_TARGET = "CPU architecture"
+   ```
+
+3. If targeting a platform not supplied by WAMR, refer to the [WAMR porting guide](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/port_wamr.md#wamr-porting-guide) and set `WAMR_BUILD_TARGET`, `WAMR_BUILD_PLATFORM` and `WAMR_SHARED_PLATFORM_CONFIG` in the `.cargo/config.toml` properly.
+
+##### For espidf targets
+
+1. Get the latest information from [The Rust on ESP Book](https://docs.esp-rs.org/book/writing-your-own-application/index.html).
+2. Generate projects from templates following the instructions in [The Rust on ESP Book](https://docs.esp-rs.org/book/writing-your-own-application/generate-project/index.html).
+3. Add the following configuration to your project's `Cargo.toml`:
+
+   ```toml
+   wamr-rust-sdk = { git = "https://github.com/bytecodealliance/wamr-rust-sdk", features = ["esp-idf"] }
+   ```
+
+Use bulding on MacOS as an example:
+
+```sh
+$ brew install espup
+$ espup install
+
+$ mkdir ~/esp
+$ git clone --recursive https://github.com/espressif/esp-idf.git ~/esp/esp-idf
+$ cd ~/esp/esp-idf && ./install.sh
+
+# in espidftemplate-with-wamr
```

Comment:
this should be replaced with the folder that the person has chosen when building his own template